shared/utility/math/matrix/transform.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2013 NUbots <nubots@nubots.net>
+ */
+
+#ifndef UTILITY_MATH_TRANSFORM_H
+#define UTILITY_MATH_TRANSFORM_H
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+#include "utility/math/angle.h"
+
+/**
+ * Matrix related helper methods
+ *
+ * @author Brendan Annable
+ */
+
+namespace utility {
+namespace math {
+    namespace transform {
+
+        inline Eigen::Affine2d worldToLocal(const Eigen::Affine2d world, const Eigen::Affine2d& reference) {
+            Eigen::Affine2d diff;
+            double angle       = utility::math::angle::normalizeAngle(Eigen::Rotation2Dd(reference.linear()).angle()
+                                                                - Eigen::Rotation2Dd(world.linear()).angle());
+            diff.linear()      = Eigen::Rotation2Dd(angle).toRotationMatrix();
+            diff.translation() = reference.translation() - world.translation();
+
+            Eigen::Affine2d result;
+            result.translation() = world.linear() * diff.translation();
+            result.linear()      = diff.linear();
+            return result;
+        }
```

Comment:
Is this all that's left of the transform? Isn't this just the same as world.inverse() * reference or something? We have been trying to get away from this world local etc with the `Htw` etc code. So this should just be some multiplication of either `world * reference` or `world.inverse() * reference`

Reply:
`worldToLocal` seems to have no connection to `world * reference` or `world.inverse() * reference`, or any permutation of those that I can think of.

Reply:
the intention of worldToLocal was that you'd pass a world reference and get it back as a local reference. Then the corresponding localToWorld would do the opposite. The idea being that the world transform you give is actually a world to local transform itself. So if it's not doing that's a bad sign

Reply:
It appears that `worldToLocal` should be equivalent to `world.inverse() * reference`. This will need further investigating once we get to a module that actually matters (like the walk engine)